internal/fs/inode/dir.go

```diff
@@ -529,6 +529,92 @@ func (d *dirInode) ReadDescendants(ctx context.Context, limit int) (map[Name]*Co
 
 }
 
+
+
+func (d *dirInode) readMinObjects(
+	ctx context.Context,
+	tok string) (cores map[Name]*MinCore, newTok string, err error) {
+	// Ask the bucket to list some objects.
+	req := &gcs.ListObjectsRequest{
+		Delimiter:                "/",
+		IncludeTrailingDelimiter: true,
+		Prefix:                   d.Name().GcsObjectName(),
+		ContinuationToken:        tok,
+		MaxResults:               MaxResultsForListObjectsCall,
+		// Setting Projection param to noAcl since fetching owner and acls are not
+		// required.
+		ProjectionVal: gcs.NoAcl,
+	}
+
+	listing, err := d.bucket.ListMinObjects(ctx, req)
+	if err != nil {
+		err = fmt.Errorf("ListObjects: %w", err)
+		return
+	}
+
+	cores = make(map[Name]*MinCore)
+	defer func() {
+		now := d.cacheClock.Now()
+		for fullName, c := range cores {
+			d.cache.Insert(now, path.Base(fullName.LocalName()), c.MinObjectType())
+		}
+	}()
+
+	for _, o := range listing.Objects {
+		// Skip empty results or the directory object backing this inode.
+		if o.Name == d.Name().GcsObjectName() || o.Name == "" {
+			continue
+		}
+
+		nameBase := path.Base(o.Name) // ie. "bar" from "foo/bar/" or "foo/bar"
+
+		// Given the alphabetical order of the objects, if a file "foo" and
+		// directory "foo/" coexist, the directory would eventually occupy
+		// the value of records["foo"].
+		if strings.HasSuffix(o.Name, "/") {
+			dirName := NewDirName(d.Name(), nameBase)
+			explicitDir := &MinCore{
+				Bucket:   d.Bucket(),
+				FullName: dirName,
+				Object:   o,
```

Comment:
can you add this minObject as new property in existing core struct defintion. That would avoid replicating all this code.